drivers/timer/arcv2_timer0.c

```diff
@@ -354,14 +354,14 @@ void sys_clock_disable(void)
 	unsigned int key;  /* interrupt lock level */
 	u32_t control; /* timer control register value */
 
-	key = irq_lock();
+	key = arch_irq_lock();
```

Comment:
Even though replacement of `irq_lock()` with `arch_irq_lock()` is harmless here as this operation makes only sense for a local CPU but my question is why do we need to globally disable IRQs on the core here? Typically we do that to exclude possibility to re-enter a critical section. But OK what happens if we re-enter this code? Or rather how is it possible to get here multiple times if not from IRQ handler, but then do we really expect IRQ handler to disable the clock???